sdk/spring/azure-spring-boot-samples/azure-spring-boot-sample-cosmos-multi-database-multi-account/src/main/java/com/azure/spring/sample/cosmos/multi/database/multiple/account/MultiDatabaseApplication.java

```diff
@@ -0,0 +1,81 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.azure.spring.sample.cosmos.multi.database.multiple.account;
+
+import com.azure.cosmos.models.PartitionKey;
+import com.azure.spring.data.cosmos.core.CosmosTemplate;
+import com.azure.spring.data.cosmos.core.ReactiveCosmosTemplate;
+import com.azure.spring.data.cosmos.repository.support.CosmosEntityInformation;
+import com.azure.spring.sample.cosmos.multi.database.multiple.account.repository.cosmos.CosmosUser;
+import com.azure.spring.sample.cosmos.multi.database.multiple.account.repository.cosmos.UserRepository;
+import com.azure.spring.sample.cosmos.multi.database.multiple.account.repository.mysql.MysqlUser;
+import com.azure.spring.sample.cosmos.multi.database.multiple.account.repository.mysql.MysqlUserRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.boot.CommandLineRunner;
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+
+import javax.annotation.PostConstruct;
+import javax.annotation.PreDestroy;
+import java.util.List;
+
+@SpringBootApplication
+public class MultiDatabaseApplication implements CommandLineRunner {
+
+    @Autowired
+    private UserRepository userRepository;
+
+    @Autowired
+    private MysqlUserRepository mysqlUserRepository;
+
+    @Autowired
+    @Qualifier("secondaryDatabaseTemplate")
+    private CosmosTemplate secondaryDatabaseTemplate;
+
+    @Autowired
+    @Qualifier("primaryDatabaseTemplate")
+    private ReactiveCosmosTemplate primaryDatabaseTemplate;
+
+    private final CosmosUser cosmosUser = new CosmosUser("1024", "1024@geek.com", "1k", "Mars");
+    private MysqlUser userForMYSQL;
+    private static CosmosEntityInformation<CosmosUser, String> userInfo = new CosmosEntityInformation<>(CosmosUser.class);
+
+    public static void main(String[] args) {
+        SpringApplication.run(MultiDatabaseApplication.class, args);
+    }
+
+    public void run(String... var1) throws Exception {
+
+        final List<CosmosUser> cosmosUsers = this.userRepository.findByEmailOrName(this.cosmosUser.getEmail(),
+            this.cosmosUser.getName()).collectList().block();
```

Comment:
Is this same to `primaryDatabaseTemplate.findById(...)`?

Reply:
nope, this function is defined in `UserRepository` which will be analysis by cosmos, the same function is `this.userRepository.findById(cosmosUser.getId()).block()`